torch/distributions/exp_family.py

```diff
@@ -58,3 +58,42 @@ def entropy(self):
         for np, g in zip(nparams, gradients):
             result -= np * g
         return result
+
+    @staticmethod
+    def _from_natural_params(*params):
+        raise NotImplementedError
+
+    def normalized_product(self, dim=-1, keepdim=False):
+        r"""
+        Returns a new distribution object whose probability mass/density function is the normalized product 
+        of the probability mass/density functions along given axis. The product of exponential family 
+        distributions results in a distribution in the same family, but with new natural parameters :math:`\sum(\theta)` 
+
+        .. math::
+
+            \frac{1}{c} \prod_i p_{F}(x; \theta_i) = \exp(\langle t(x), \sum_i\theta_i\rangle - F(\sum_i\theta_i) + k(x))
+
+        where c is the normalization constant and i is the index along given dimension.
+
+        For a batched distribution d and a single observation x::
+
+        d.prod(dim).log_prob(x) = d.log_prob(x).sum(dim) + c
+
+        This might be useful for models such as Product-of-Experts (Hinton, 1999):
+
+        .. math::
+
+            p(\textbf{d}; \theta_1...\theta_n) = \frac{\prod_m p_m(\textbf{d}|\theta_m)}{\sum_i\prod_m p_m(\textbf{c}_i|\theta_m)}
+
+        where d is a data vector in a discrete space, m is all the parameters of individual model m, 
```

Comment:
```:math:`\textbf{d}` is a data vector in a discrete space, :math:`m` is all the parameters of the individual model M```

(also, I think it's better to call the model M and the parameters m)

Reply:
There was a typo, m -> theta_m. Fixed now. Final thing is

```rst
where :math:`\textbf{d}` is a data vector in a discrete space, :math:`\theta_m` is all the parameters of individual model m, :math:`p_m(\textbf{d}|\theta_m)` is the probability of :math:`\textbf{d}` under model :math:`m`, and :math:`i` is an index over all possible vectors in the data space.
```